Testing/UnitTests/xASL_ut_UnitTest_function_str2num.m

```diff
@@ -0,0 +1,102 @@
+function UnitTest = xASL_ut_UnitTest_function_str2num(TestRepository)
+%xASL_ut_UnitTest_function_str2num Individual unit test for xASL_str2num
+%
+% INPUT:        TestRepository - Path to test repository.
+%
+% OUTPUT:       UnitTest  - Test structure
+%               name      - Name of tested module or submodule (char array)
+%               unit      - Insert one of the following: 'Module', 'Submodule' or 'Function'
+%               passed    - Result of all subtests combined (true or false)
+%               test      - Structure with individual subtest results
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:  Should be run using xASL_ut_UnitTesting.
+%
+% EXAMPLE:      UnitTests(1) = xASL_ut_UnitTest_function_str2num(TestRepository);
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% Copyright 2015-2021 ExploreASL
+
+%% Initialize test structure
+
+% Insert test name here
+UnitTest.name = 'xASL_str2num';
+
+% Define whether you are testing a module, submodule or function
+UnitTest.unit = 'Function';
+
+%% Test run 1
+
+% Give your individual subtest a name
+UnitTest.tests(1).testname = 'Number types: integer, double, positive and negative';
+
+% Start the test
+testTime = tic;
+
+% Run your test here
+numOutA = xASL_str2num('123');
+numOutB = xASL_str2num('-123');
+numOutC = xASL_str2num('123.456');
```

Comment:
Same suggestion here as for the other unit test below, why do you test `123456`, why not `1234567890`? But that's just my theoretical thinking probably :)

Reply:
From the scientific point of view this is definitely a valid argument, but on the other hand we can't test for every integer/double value. In my opinion testing with one positive integer, one negative interger and a double should be enough. We can write something like...

```
index = 1;
for iTest = -1000:+1000
    testResult(index) = xASL_str2num(num2str(iTest));
    index = index +1;
end
```

... though.

Reply:
I don't expect relevant testing results here though. What could be more interesting is to test the cases where we have problems regarding rounding a bit better. So tests like:

```
numOutX = xASL_str2num('0.000000000001');
numOutY = xASL_str2num('-0.000000000001');
```